homeassistant/components/irm_kmi/config_flow.py

```diff
@@ -0,0 +1,147 @@
+"""Config flow to set up IRM KMI integration via the UI."""
+
+import asyncio
+import logging
+
+from irm_kmi_api import IrmKmiApiClient
+import voluptuous as vol
+
+from homeassistant.components.zone import DOMAIN as ZONE_DOMAIN
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult, OptionsFlow
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, CONF_ZONE
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    EntitySelector,
+    EntitySelectorConfig,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import (
+    CONF_LANGUAGE_OVERRIDE,
+    CONF_LANGUAGE_OVERRIDE_OPTIONS,
+    CONFIG_FLOW_VERSION,
+    DOMAIN,
+    OUT_OF_BENELUX,
+    USER_AGENT,
+)
+from .types import IrmKmiConfigEntry
+from .utils import get_config_value
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IrmKmiConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Configuration flow for the IRM KMI integration."""
+
+    VERSION = CONFIG_FLOW_VERSION
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: IrmKmiConfigEntry) -> OptionsFlow:
+        """Create the options flow."""
+        return IrmKmiOptionFlow(config_entry)
+
+    async def async_step_user(self, user_input: dict | None = None) -> ConfigFlowResult:
+        """Define the user step of the configuration flow."""
+        errors = {}
+
+        if user_input:
+            _LOGGER.debug("Provided config user is: %s", user_input)
+
+            if (zone := self.hass.states.get(user_input[CONF_ZONE])) is None:
+                errors[CONF_ZONE] = "zone_not_exist"
```

Comment:
why do we use a zone instead of lang lot?

Reply:
I assume you meant "instead of latitude longitude".  Most of the time, users want weather for their home (which they already configured when they set up Home Assistant) so it seemed natural to pick a zone.  If they want weather for another location, they can create another zone and set up the integration with that one.  

Do you think it should be changed to latitude and longitude in the configuration flow?

Reply:
You can do it like other weather integrations and use a LocationSelector with default position set to the home location: https://github.com/home-assistant/core/blob/6de2258325bfb22ef27d2da1f8d6a0cac607227b/homeassistant/components/smhi/config_flow.py#L67-L77
That will lower the complexity as you don't have moving zones anymore and therefore no repair issues.

Reply:
Indeed, this would remove the need for repairs.  I'll look into it and implement the configuration flow using latitude and longitude instead of a zone.  Thanks for the pointer to another component

Reply:
This is implemented in commit 1ea8bb0.